### PR TITLE
DEV-3873: Add interval filtering to portal contributions

### PR DESCRIPTION
**Pull request by GuiMend**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
main <- [DEV-3873 <- DEV-3874](https://github.com/newsrevenuehub/rev-engine/pull/1494)
#### What's this PR do?
- Add filter based on interval type for contributions
#### Why are we doing this? How does it help us?
- Enable user to filter contributions by "All", "One-time", "Recurring"
#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?
yes
#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
no
#### Have automated unit tests been added? If not, why?
yes
#### How should this change be communicated to end users?
n/a
#### Are there any smells or added technical debt to note?
no
#### Has this been documented? If so, where?
no
#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-3873
#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:
no
#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
https://news-revenue-hub.atlassian.net/browse/DEV-3874

### CODE REVIEW

File: apps/contributions/filters.py [hunk 1]
```diff
@@ -60,13 +61,21 @@ def filter_queryset_by_rp(self, queryset, revenue_program_id: str):
             | Q(contribution_metadata__contains={"revenue_program_id": revenue_program_id})
         )
 
+    def filter_intervals(self, queryset, request):
+        if (interval := request.query_params.get("interval")) == ContributionInterval.ONE_TIME.value:
+            return queryset.filter(interval=ContributionInterval.ONE_TIME)
+        elif interval == "recurring":
+            return queryset.exclude(interval=ContributionInterval.ONE_TIME)
+        return queryset
+
```
Comment (benjaminEwhite) [inline apps/contributions/filters.py:70]: Minor suggestion, but this is a good place for match/case:

```suggestion
        match request.query_params.get("interval"):
            case ContributionInterval.ONE_TIME.value:
                return queyrset.filter(interval=ContributionInterval.ONE_TIME)
            case "recurring":
                return queryset.exclude(ContributionInterval.ONE_TIME)
            case _:
                return queryset
```

Comment (benjaminEwhite) [inline apps/contributions/filters.py:70]: Minor suggestion, but this is a good place for match/case:

```suggestion
        match request.query_params.get("interval"):
            case ContributionInterval.ONE_TIME.value:
                return queyrset.filter(interval=ContributionInterval.ONE_TIME)
            case "recurring":
                return queryset.exclude(ContributionInterval.ONE_TIME)
            case _:
                return queryset
```

Comment (benjaminEwhite) [reply apps/contributions/filters.py:70]: Consider putting "recurring" into a constant at top of file (all caps), and then can import that var in tests.

File: apps/contributions/tests/test_filters.py [hunk 2]
```diff
@@ -29,6 +29,48 @@ def contributions(self, valid_metadata_factory):
...
+
+        if interval_option == "one_time":
+            assert one_time in filtered
+            assert yearly not in filtered
+            assert monthly not in filtered
+        elif interval_option == "recurring":
+            assert yearly in filtered
+            assert monthly in filtered
+            assert one_time not in filtered
+        else:
```
Comment (benjaminEwhite) [inline apps/contributions/tests/test_filters.py:69]: Also good to use match/case here.

**Review by benjaminEwhite — CHANGES_REQUESTED** [review]

File: apps/contributions/filters.py (hunk 1 above)
Comment (GuiMend) [reply apps/contributions/filters.py:70]: I tried adding the variable at the top of the file but when trying to use it inside of the match/case it didn't work.
image

I was able to make it work by adding it inside `PortalContributionFilter`. Is this a good workaround?
image
image

**Review by benjaminEwhite — APPROVED** [review]
LGTM

**Pull request merged**